conn.go

```diff
@@ -488,3 +500,59 @@ type TraceTag struct {
 	// Action - specifies an action, such as an INSERT or UPDATE operation, in a module
 	Action string
 }
+
+func (c *conn) openProxyConnection(ctx context.Context) error {
+	if !c.connParams.HeterogeneousPool {
+		return nil
+	}
+
+	var user string
+	var ok bool
+	if user, ok = ctx.Value(ProxyUser).(string); ok && user == c.proxyUser {
+		return nil
+	}
+
+	if c.dpiConn != nil {
+		if err := c.Close(); err != nil {
+			return driver.ErrBadConn
+		}
+	}
+
+	c.Lock()
+	defer c.Unlock()
+
+	var connCreateParams C.dpiConnCreateParams
```

Comment:
This should be refactored to share the common parts with openConn.

Reply:
What's the best way to reuse openConn here? This function return a new conn struct.

Reply:
I created the acquireConn method to share commom parts of openConn with ensureContextUser.